x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -12,6 +12,7 @@ use thiserror::Error;
 /// Defer deserialization from bytes
 /// <div class="warning">It is possible that this type performs multiple allocations during deserialization. It should be used sparingly.</div>
 #[cfg_attr(feature = "debug", derive(Debug))]
+#[derive(BorshSerialize)]
```

Comment:
Necessary to be able to return a `DeferDeserialize` from the `execute` function. Let me know if it's unneeded, we could just have returned a `Vec<u8>` and casted to a `DeferDeserialize` in the test directly.

Reply:
I will remove it because it really looks unnecessary

Reply:
I was wondering if I should remove it, it should probably be left like this at the end because casting the return bytes to a `DeferDeserialize` is necessary, otherwise the deserialization expects a length prefix if we wanted bytes such as `Vec<u8>`. Further, the `BorshSerialize` saves us from a manual `into_inner` function that would just be
```rust
impl DeferDeserialize {
    ...

    pub fn into_inner(self) -> Vec<u8> {
        self.0
    }
}
```